Integration tests are failing on this PR, but I'm not sure what the status of these integration tests is. I made a test PR against main (#1107) where the only change is adding my name to `CONTRIBUTORS`, and integration tests fail there too. Seven tests fail in each PR, one is different in each. I've run the integration tests twice on each PR.

Six failures are in common across both PRs (both runs):
```
FAILED tests/integration/test_application.py::test_upgrade_local_charm - juju...
FAILED tests/integration/test_model.py::test_deploy_bundle_with_overlay_as_argument
FAILED tests/integration/test_model.py::test_deploy_bundle_with_multiple_overlays_with_include_files
FAILED tests/integration/test_application.py::test_app_relation_destroy_block_until_done
FAILED tests/integration/test_model.py::test_wait_for_idle_with_not_enough_units
FAILED tests/integration/test_model.py::test_wait_for_idle_more_units_than_needed
```

Unique to this PR (one of the two runs only):
```
FAILED tests/integration/test_model.py::test_unit_annotations - asyncio.excep...
```

Unique to dummy PR [#1107] (both runs):
```
FAILED tests/integration/test_application.py::test_upgrade_local_charm_resource
```

I'm going to rerun the integration tests and see if they have the same results.

--- a/setup.py
+++ b/setup.py
@@ -31,7 +31,7 @@ setup(
         'pyasn1>=0.4.4',
         'toposort>=1.5,<2',
         'typing_inspect>=0.6.0',
-        'kubernetes>=12.0.1',
+        'kubernetes>=12.0.1,<31.0.0',
         'hvac',
         'packaging',
     ],

--- a/tox.ini
+++ b/tox.ini
@@ -38,7 +38,7 @@ deps =
     pytest-xdist
     Twine
     websockets
-    kubernetes
+    kubernetes<31.0.0
     hvac
     packaging
     setuptools

